Go panic in OLTP exporter

**Describe the bug**
Go is periodically producing a panic log and restarting (with an exit code of 2) with one of 2 outputs:

````
panic: runtime error: index out of range [-2]

goroutine 112 [running]:
go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/metrics/v1.encodeVarintMetricsService(...)
	go.opentelemetry.io/collector/pdata@v0.63.1/internal/data/protogen/collector/metrics/v1/metrics_service.pb.go:438
go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/metrics/v1.(*ExportMetricsServiceRequest).MarshalToSizedBuffer(0xc011b58138, {0xc018800000, 0xc666, 0xc666})
	go.opentelemetry.io/collector/pdata@v0.63.1/internal/data/protogen/collector/metrics/v1/metrics_service.pb.go:357 +0x16d
go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/metrics/v1.(*ExportMetricsServiceRequest).Marshal(0xc01dab1f00?)
	go.opentelemetry.io/collector/pdata@v0.63.1/internal/data/protogen/collector/metrics/v1/metrics_service.pb.go:332 +0x56
google.golang.org/protobuf/internal/impl.legacyMarshal({{}, {0x74909f8, 0xc01dab1f00}, {0x0, 0x0, 0x0}, 0x0})
	google.golang.org/protobuf@v1.28.1/internal/impl/legacy_message.go:402 +0xa2
google.golang.org/protobuf/proto.MarshalOptions.marshal({{}, 0x28?, 0x0, 0x0}, {0x0, 0x0, 0x0}, {0x74909f8, 0xc01dab1f00})
	google.golang.org/protobuf@v1.28.1/proto/encode.go:166 +0x27b
google.golang.org/protobuf/proto.MarshalOptions.MarshalAppend({{}, 0x60?, 0x3e?, 0x4c?}, {0x0, 0x0, 0x0}, {0x7410c60?, 0xc01dab1f00?})
	google.golang.org/protobuf@v1.28.1/proto/encode.go:125 +0x79
github.com/golang/protobuf/proto.marshalAppend({0x0, 0x0, 0x0}, {0x7faa4a6238e0?, 0xc011b58138?}, 0x0?)
	github.com/golang/protobuf@v1.5.2/proto/wire.go:40 +0xa5
github.com/golang/protobuf/proto.Marshal(...)
	github.com/golang/protobuf@v1.5.2/proto/wire.go:23
google.golang.org/grpc/encoding/proto.codec.Marshal({}, {0x64c3e60, 0xc011b58138})
	google.golang.org/grpc@v1.50.1/encoding/proto/proto.go:45 +0x4e
google.golang.org/grpc.encode({0x7faa4a7f6e88?, 0xb312e00?}, {0x64c3e60?, 0xc011b58138?})
	google.golang.org/grpc@v1.50.1/rpc_util.go:594 +0x44
google.golang.org/grpc.prepareMsg({0x64c3e60?, 0xc011b58138?}, {0x7faa4a7f6e88?, 0xb312e00?}, {0x0, 0x0}, {0x744f500, 0xc0000faaf0})
	google.golang.org/grpc@v1.50.1/stream.go:1692 +0xd2
google.golang.org/grpc.(*clientStream).SendMsg(0xc01827c360, {0x64c3e60?, 0xc011b58138})
	google.golang.org/grpc@v1.50.1/stream.go:830 +0xfd
google.golang.org/grpc.invoke({0x745e5f0?, 0xc011627020?}, {0x694300b?, 0x4?}, {0x64c3e60, 0xc011b58138}, {0x64c3fa0, 0xc016bd6810}, 0x0?, {0xc01bb72440, ...})
	google.golang.org/grpc@v1.50.1/call.go:70 +0xa8
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.UnaryClientInterceptor.func1({0x745e5f0, 0xc011626f60}, {0x694300b, 0x3f}, {0x64c3e60, 0xc011b58138}, {0x64c3fa0, 0xc016bd6810}, 0xc000c40380, 0x6a6f038, ...)
	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.36.4/interceptor.go:105 +0x3e4
google.golang.org/grpc.(*ClientConn).Invoke(0xc000c40380?, {0x745e5f0?, 0xc011626f60?}, {0x694300b?, 0x3f?}, {0x64c3e60?, 0xc011b58138?}, {0x64c3fa0?, 0xc016bd6810?}, {0xc0003dded0, ...})
	google.golang.org/grpc@v1.50.1/call.go:35 +0x223
go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/metrics/v1.(*metricsServiceClient).Export(0xc00013e458, {0x745e5f0, 0xc011626f60}, 0xc000a02e70?, {0xc0003dded0, 0x1, 0x1})
	go.opentelemetry.io/collector/pdata@v0.63.1/internal/data/protogen/collector/metrics/v1/metrics_service.pb.go:272 +0xc9
go.opentelemetry.io/collector/pdata/pmetric/pmetricotlp.(*grpcClient).Export(0x49cd20?, {0x745e5f0?, 0xc011626f60?}, {0xc011626f30?}, {0xc0003dded0?, 0x0?, 0x0?})
	go.opentelemetry.io/collector/pdata@v0.63.1/pmetric/pmetricotlp/grpc.go:50 +0x30
go.opentelemetry.io/collector/exporter/otlpexporter.(*exporter).pushMetrics(0xc00131c820, {0x745e5b8?, 0xc00f8ecd20?}, {0x745e5f0?})
	go.opentelemetry.io/collector/exporter/otlpexporter@v0.63.1/otlp.go:111 +0x69
go.opentelemetry.io/collector/exporter/exporterhelper.(*metricsRequest).Export(0x745e5f0?, {0x745e5b8?, 0xc00f8ecd20?})
	go.opentelemetry.io/collector@v0.63.1/exporter/exporterhelper/metrics.go:65 +0x34
go.opentelemetry.io/collector/exporter/exporterhelper.(*timeoutSender).send(0xc001324840, {0x747d640, 0xc0123cf4a0})
	go.opentelemetry.io/collector@v0.63.1/exporter/exporterhelper/common.go:203 +0x96
go.opentelemetry.io/collector/exporter/exporterhelper.(*retrySender).send(0xc0005fd440, {0x747d640, 0xc0123cf4a0})
	go.opentelemetry.io/collector@v0.63.1/exporter/exporterhelper/queued_retry.go:388 +0x58d
go.opentelemetry.io/collector/exporter/exporterhelper.(*metricsSenderWithObservability).send(0xc00131f668, {0x747d640, 0xc0123cf4a0})
	go.opentelemetry.io/collector@v0.63.1/exporter/exporterhelper/metrics.go:133 +0x88
go.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).start.func1({0x747d640, 0xc0123cf4a0})
	go.opentelemetry.io/collector@v0.63.1/exporter/exporterhelper/queued_retry.go:206 +0x39
go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*boundedMemoryQueue).StartConsumers.func1()
	go.opentelemetry.io/collector@v0.63.1/exporter/exporterhelper/internal/bounded_memory_queue.go:61 +0xb6
created by go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*boundedMemoryQueue).StartConsumers
	go.opentelemetry.io/collector@v0.63.1/exporter/exporterhelper/internal/bounded_memory_queue.go:56 +0x45
````

````
panic: runtime error: index out of range [-1]

goroutine 125 [running]:
go.opentelemetry.io/collector/pdata/internal/data/protogen/metrics/v1.(*ResourceMetrics).MarshalToSizedBuffer(0xc01f3f3500, {0xc012466000, 0x5d4, 0x11790})
	go.opentelemetry.io/collector/pdata@v0.63.1/internal/data/protogen/metrics/v1/metrics.pb.go:2144 +0x1d6
go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/metrics/v1.(*ExportMetricsServiceRequest).MarshalToSizedBuffer(0xc019f38150, {0xc012466000, 0x11790, 0x11790})
	go.opentelemetry.io/collector/pdata@v0.63.1/internal/data/protogen/collector/metrics/v1/metrics_service.pb.go:352 +0xac
go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/metrics/v1.(*ExportMetricsServiceRequest).Marshal(0xc01d6cc330?)
	go.opentelemetry.io/collector/pdata@v0.63.1/internal/data/protogen/collector/metrics/v1/metrics_service.pb.go:332 +0x56
google.golang.org/protobuf/internal/impl.legacyMarshal({{}, {0x74909f8, 0xc01d6cc330}, {0x0, 0x0, 0x0}, 0x0})
	google.golang.org/protobuf@v1.28.1/internal/impl/legacy_message.go:402 +0xa2
google.golang.org/protobuf/proto.MarshalOptions.marshal({{}, 0x28?, 0x0, 0x0}, {0x0, 0x0, 0x0}, {0x74909f8, 0xc01d6cc330})
	google.golang.org/protobuf@v1.28.1/proto/encode.go:166 +0x27b
google.golang.org/protobuf/proto.MarshalOptions.MarshalAppend({{}, 0x60?, 0x3e?, 0x4c?}, {0x0, 0x0, 0x0}, {0x7410c60?, 0xc01d6cc330?})
	google.golang.org/protobuf@v1.28.1/proto/encode.go:125 +0x79
github.com/golang/protobuf/proto.marshalAppend({0x0, 0x0, 0x0}, {0x7f836b82f3d0?, 0xc019f38150?}, 0x0?)
	github.com/golang/protobuf@v1.5.2/proto/wire.go:40 +0xa5
github.com/golang/protobuf/proto.Marshal(...)
	github.com/golang/protobuf@v1.5.2/proto/wire.go:23
google.golang.org/grpc/encoding/proto.codec.Marshal({}, {0x64c3e60, 0xc019f38150})
	google.golang.org/grpc@v1.50.1/encoding/proto/proto.go:45 +0x4e
google.golang.org/grpc.encode({0x7f836bc31170?, 0xb312e00?}, {0x64c3e60?, 0xc019f38150?})
	google.golang.org/grpc@v1.50.1/rpc_util.go:594 +0x44
google.golang.org/grpc.prepareMsg({0x64c3e60?, 0xc019f38150?}, {0x7f836bc31170?, 0xb312e00?}, {0x0, 0x0}, {0x744f500, 0xc000150aa0})
	google.golang.org/grpc@v1.50.1/stream.go:1692 +0xd2
google.golang.org/grpc.(*clientStream).SendMsg(0xc0192d3200, {0x64c3e60?, 0xc019f38150})
	google.golang.org/grpc@v1.50.1/stream.go:830 +0xfd
google.golang.org/grpc.invoke({0x745e5f0?, 0xc01bacc840?}, {0x694300b?, 0x4?}, {0x64c3e60, 0xc019f38150}, {0x64c3fa0, 0xc01c067aa0}, 0x0?, {0xc01db78340, ...})
	google.golang.org/grpc@v1.50.1/call.go:70 +0xa8
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.UnaryClientInterceptor.func1({0x745e5f0, 0xc01bacc780}, {0x694300b, 0x3f}, {0x64c3e60, 0xc019f38150}, {0x64c3fa0, 0xc01c067aa0}, 0xc0002da380, 0x6a6f038, ...)
	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.36.4/interceptor.go:105 +0x3e4
google.golang.org/grpc.(*ClientConn).Invoke(0xc0002da380?, {0x745e5f0?, 0xc01bacc780?}, {0x694300b?, 0x3f?}, {0x64c3e60?, 0xc019f38150?}, {0x64c3fa0?, 0xc01c067aa0?}, {0xc00081d0d0, ...})
	google.golang.org/grpc@v1.50.1/call.go:35 +0x223
go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/metrics/v1.(*metricsServiceClient).Export(0xc000012fa8, {0x745e5f0, 0xc01bacc780}, 0xc001318750?, {0xc00081d0d0, 0x1, 0x1})
	go.opentelemetry.io/collector/pdata@v0.63.1/internal/data/protogen/collector/metrics/v1/metrics_service.pb.go:272 +0xc9
go.opentelemetry.io/collector/pdata/pmetric/pmetricotlp.(*grpcClient).Export(0x49cd20?, {0x745e5f0?, 0xc01bacc780?}, {0xc01bacc750?}, {0xc00081d0d0?, 0x0?, 0x0?})
	go.opentelemetry.io/collector/pdata@v0.63.1/pmetric/pmetricotlp/grpc.go:50 +0x30
go.opentelemetry.io/collector/exporter/otlpexporter.(*exporter).pushMetrics(0xc001193c20, {0x745e5b8?, 0xc01605af60?}, {0x745e5f0?})
	go.opentelemetry.io/collector/exporter/otlpexporter@v0.63.1/otlp.go:111 +0x69
go.opentelemetry.io/collector/exporter/exporterhelper.(*metricsRequest).Export(0x745e5f0?, {0x745e5b8?, 0xc01605af60?})
	go.opentelemetry.io/collector@v0.63.1/exporter/exporterhelper/metrics.go:65 +0x34
go.opentelemetry.io/collector/exporter/exporterhelper.(*timeoutSender).send(0xc0008e0fd0, {0x747d640, 0xc01a46b4d0})
	go.opentelemetry.io/collector@v0.63.1/exporter/exporterhelper/common.go:203 +0x96
go.opentelemetry.io/collector/exporter/exporterhelper.(*retrySender).send(0xc001195170, {0x747d640, 0xc01a46b4d0})
	go.opentelemetry.io/collector@v0.63.1/exporter/exporterhelper/queued_retry.go:388 +0x58d
go.opentelemetry.io/collector/exporter/exporterhelper.(*metricsSenderWithObservability).send(0xc000d33290, {0x747d640, 0xc01a46b4d0})
	go.opentelemetry.io/collector@v0.63.1/exporter/exporterhelper/metrics.go:133 +0x88
go.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).start.func1({0x747d640, 0xc01a46b4d0})
	go.opentelemetry.io/collector@v0.63.1/exporter/exporterhelper/queued_retry.go:206 +0x39
go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*boundedMemoryQueue).StartConsumers.func1()
	go.opentelemetry.io/collector@v0.63.1/exporter/exporterhelper/internal/bounded_memory_queue.go:61 +0xb6
created by go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*boundedMemoryQueue).StartConsumers
	go.opentelemetry.io/collector@v0.63.1/exporter/exporterhelper/internal/bounded_memory_queue.go:56 +0x45
````

**Steps to reproduce**

I've only noticed this in logs recently trying to understand why I'm seeing duplicate logs being published via OLTP end so I do not have reproduction or what part of my configuration could be causing this

**What version did you use?**
0.63.1

**What config did you use?**
````yaml
exporters:
  logging:
    loglevel: warn
  otlp/newrelic:
    endpoint: ${NEW_RELIC_ENDPOINT}
    headers:
      api-key: ${NEW_RELIC_API_KEY}
  prometheus:
    endpoint: 0.0.0.0:8787
extensions:
  health_check: {}
  memory_ballast:
    size_mib: "204"
processors:
  batch:
    send_batch_size: 1000
  cumulativetodelta:
    include:
      match_type: strict
      metrics:
      - system.network.io
      - system.disk.operations
      - system.network.dropped
      - system.network.packets
      - process.cpu.time
  filter/dropMetrics:
    metrics:
      exclude:
        match_type: strict
        metric_names:
        - k8s.pod.filesystem.available
        - k8s.pod.filesystem.capacity
        - k8s.pod.filesystem.usage
  k8sattributes:
    extract:
      metadata:
      - k8s.pod.name
      - k8s.pod.uid
      - k8s.deployment.name
      - k8s.namespace.name
    filter:
      node_from_env_var: KUBE_NODE_NAME
    passthrough: false
    pod_association:
    - sources:
      - from: resource_attribute
        name: k8s.pod.uid
  memory_limiter:
    check_interval: 5s
    limit_mib: 768
    spike_limit_mib: 256
  resource:
    attributes:
    - action: insert
      key: k8s.cluster.name
      value: uksouth
    - action: upsert
      from_attribute: service.instance.id
      key: k8s.pod.uid
    - action: upsert
      key: host.id
      value: ${KUBE_NODE_NAME}
    - action: upsert
      key: host.name
      value: ${KUBE_NODE_NAME}
  resource/serviceFromK8S:
    attributes:
    - action: upsert
      from_attribute: k8s.deployment.name
      key: service.name
    - action: upsert
      from_attribute: k8s.namespace.name
      key: service.namespace
receivers:
  filelog:
    exclude:
    - /var/log/pods/apm_opentelemetry-collector*_*/opentelemetry-collector/*.log
    include:
    - /var/log/pods/apps*/*/*.log
    - /var/log/pods/kured*/*/*.log
    include_file_name: false
    include_file_path: true
    operators:
    - id: extract_metadata_from_filepath
      parse_from: attributes["log.file.path"]
      regex: ^.*/(?P<namespace_name>[^_]+)_(?P<pod_name>[^_]+)_(?P<pod_uid>[a-f0-9-]{36})/(?P<container_name>[^._]+)/\d+.log$
      type: regex_parser
    - id: parser-containerd
      parse_from: body
      regex: ^(?P<time>[^ ^Z]+Z) (stdout|stderr) ([^ ]*) ?(?P<log>.*)$
      timestamp:
        layout: '%Y-%m-%dT%H:%M:%S.%LZ'
        parse_from: attributes.time
      type: regex_parser
    - from: attributes.log
      id: move_log_to_body
      to: body
      type: move
    - from: attributes.container_name
      id: move-container-name
      to: resource["service.name"]
      type: move
    - from: attributes.namespace_name
      id: move-namespace
      to: resource["service.namespace"]
      type: move
    - from: attributes.pod_uid
      id: move-pod-uid-to-service-instance
      to: resource["service.instance.id"]
      type: move
    - id: parse-serilog-json
      if: body matches "^{.*@m.*}$"
      parse_from: body
      scope_name:
        parse_from: attributes.SourceContext
      severity:
        parse_from: attributes.@l
      timestamp:
        layout: ms
        layout_type: epoch
        parse_from: attributes.@t
      trace:
        span_id:
          parse_from: attributes.SpanId
        trace_id:
          parse_from: attributes.TraceId
      type: json_parser
    - from: attributes.@m
      id: move-message
      if: attributes["@m"] != nil
      to: body
      type: move
    start_at: beginning
  hostmetrics:
    collection_interval: 30s
    scrapers:
      cpu:
        metrics:
          system.cpu.utilization:
            enabled: "true"
      memory:
        metrics:
          system.memory.utilization:
            enabled: "true"
      network: null
  k8s_events: {}
  kubeletstats:
    auth_type: serviceAccount
    collection_interval: 30s
    endpoint: https://${KUBE_NODE_NAME}:10250
    insecure_skip_verify: true
    metric_groups:
    - pod
  otlp:
    protocols:
      grpc:
        endpoint: 0.0.0.0:4317
      http:
        endpoint: 0.0.0.0:4318
  prometheus:
    config:
      scrape_configs:
      - job_name: k8s
        kubernetes_sd_configs:
        - role: pod
        relabel_configs:
        - action: keep
          regex: "true"
          source_labels:
          - __meta_kubernetes_pod_annotation_prometheus_io_scrape
        - regex: (.*:)d+;(d+)
          replacement: $1$2
          source_labels:
          - __address__
          - __meta_kubernetes_pod_annotation_prometheus_io_port
          target_label: __address__
service:
  extensions:
  - health_check
  - memory_ballast
  pipelines:
    logs:
      exporters:
      - otlp/newrelic
      processors:
      - memory_limiter
      - batch
      - resource
      - k8sattributes
      - resource/serviceFromK8S
      receivers:
      - filelog
      - k8s_events
    metrics:
      exporters:
      - otlp/newrelic
      - prometheus
      processors:
      - memory_limiter
      - filter/dropMetrics
      - batch
      - resource
      - k8sattributes
      - resource/serviceFromK8S
      - cumulativetodelta
      receivers:
      - otlp
      - hostmetrics
      - kubeletstats
      - prometheus
    traces:
      exporters:
      - otlp/newrelic
      processors:
      - memory_limiter
      - batch
      - resource
      receivers:
      - otlp
  telemetry:
    metrics:
      address: 0.0.0.0:8888

````

**Environment**
Helm chart: 0.38.0
Azure Kubernetes Service: v1.22.11

**Additional context**
Add any other context about the problem here.

--- a/.chloggen/fixprommuta.yaml
+++ b/.chloggen/fixprommuta.yaml
@@ -1,0 +1,12 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: prometheusexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Mark prometheus exporter as mutable since it sorts attributes
+
+# One or more tracking issues related to the change
+issues: [16499]
+

--- a/exporter/prometheusexporter/factory.go
+++ b/exporter/prometheusexporter/factory.go
@@ -68,7 +68,8 @@ func createMetricsExporter(
 		set,
 		cfg,
 		prometheus.ConsumeMetrics,
-		exporterhelper.WithCapabilities(consumer.Capabilities{MutatesData: false}),
+		// TODO: Consider to revert to non mutable data, need to not modify the incoming data, like sorting when calculating the signature.
+		exporterhelper.WithCapabilities(consumer.Capabilities{MutatesData: true}),
 		exporterhelper.WithStart(prometheus.Start),
 		exporterhelper.WithShutdown(prometheus.Shutdown),
 	)

